### PR TITLE
Allows exportDomainsWithSettingsXML command with no file path

### PR DESCRIPTION
**Pull request by louizo**
Currently, the exportDomainsWIthSettingsXML command expects the file path
to be passed as parameter, as an absolute path.

The default file path mentioned in ParameterFrameworkConfiguration.xml file
is used if there is no command argument. If this path is a relative one,
it is converted into absolute path before being provided to
ParameterManager.

Signed-off-by: Thierry Quervel <thierry.quervel@intel.com>

Conflicts:
	parameter/ParameterMgr.cpp
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/louizo%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12612524%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/251%23issuecomment-143169846%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-09-25T09%3A30%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12612524%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/louizo%22%7D%7D%2C%20%7B%22body%22%3A%20%22quick%20test%20%26%20working%20fine%20on%20linux.%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-09-25T10%3A50%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12612524%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/louizo%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-28T09%3A32%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%209b9c0676a694d9f3fe64eabcc3cd491b4c0a832e%20parameter/ParameterMgr.cpp%2038%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/251%23discussion_r40528778%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22After%20rebase%5Cr%5Cn%20CXmlDocSource%3A%3AmkURI%28_strXmlConfigurationUri%2C%20pConfigurableDomainsFileLocation-%3EgetUri%28%29%29%22%2C%20%22created_at%22%3A%20%222015-09-28T08%3A40%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterMgr.cpp%3AL1503-1538%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/251%23issuecomment-143169846%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/251%23issuecomment-143185997%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/251%23discussion_r40528778%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/251%23issuecomment-143690781%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/louizo'></a>

- [ ] <a href='#crh-comment-Pull 9b9c0676a694d9f3fe64eabcc3cd491b4c0a832e parameter/ParameterMgr.cpp 38'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/251#discussion_r40528778'>File: parameter/ParameterMgr.cpp:L1503-1538</a></b>
- <a href='https://github.com/miguelgaio'></a> After rebase
CXmlDocSource::mkURI(_strXmlConfigurationUri, pConfigurableDomainsFileLocation->getUri())
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/251#issuecomment-143169846'>General Comment</a></b>
- <a href='https://github.com/louizo'></a> :shipit:
- <a href='https://github.com/louizo'></a> quick test & working fine on linux.

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/251?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/251?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/251?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/251'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/251'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by louizo** [comment]
:shipit:

**Comment by louizo** [comment]
quick test & working fine on linux.

File: parameter/ParameterMgr.cpp [hunk 1]
```diff
@@ -1503,7 +1503,35 @@ CParameterMgr::CCommandHandler::CommandStatus
...
+	        pParameterConfigurationGroup->findChildOfKind("ConfigurableDomainsFileLocation"));
+
+        if (!pConfigurableDomainsFileLocation) {
+            strResult = "No ConfigurableDomainsFileLocation element found for SystemClass " +
+	        getSystemClass()->getName();
+            return CCommandHandler::EFailed;
+        }
+
+        // Get Xml configuration domains file name
+        strFileName = pConfigurableDomainsFileLocation->getFilePath(_strXmlConfigurationFolderPath);
```
Comment (miguelgaio) [inline parameter/ParameterMgr.cpp]: After rebase
 CXmlDocSource::mkURI(_strXmlConfigurationUri, pConfigurableDomainsFileLocation->getUri())

**Comment by miguelgaio** [comment]
:+1:

**Comment by krocard** [comment]
@louizo Please resubmit against `serialization_api` branch.

**Comment by krocard** [comment]
Resubmited as #284 against `master` branch as it have nothing to do with the serialization_api. Thus removing the `resubmit` label.

File: parameter/ParameterMgr.cpp (hunk 1 above)
Comment (krocard) [inline parameter/ParameterMgr.cpp]: :+1: done in #284

**Pull request closed without merging**